### ISSUE TITLE
Don't use Unity version for pub.dev versioning - breaks semantic versioning

### ISSUE THREAD

**Issue by jamesncl** [juicycleff/flutter-unity-view-widget#593]
Hi, I like the motivation behind the idea to use the supported Unity version as the pub.dev version (eg `2022.1.0+5`) but this causes significant problems as it goes against [the semantic versioning system](https://semver.org/) because you are introducing breaking changes when according to the version number you release, it should be a non-breaking patch version.

For example, from version `2022.1.0+4` to version `2022.1.0+5` there is a major breaking change - upgrading to Flutter 3, updating the Dart SDK requirement, and this has also introduced a significant problem which you have documented regarding AndroidView. Such a change should only occur when the MAJOR (first) number changes. As it stands, anyone who uses:

```
dependencies:
  flutter_unity_widget: ^2022.1.0+x
```
where `x` is any number, they will automatically update to the latest version at build time any time there is a change, potentially introducing unexpected changes and broken apps.

It would be much better to go back to the proper semantic versioning and use a simple table to document which version of Unity is supported, for example:

```
| Version     | Supported Unity version 
----------------------------------------
| 4.2.5    | 2022.1.0
| 4.2.4    | 2022.1.0
... etc
```
Hope this helps, love the package as it's incredibly useful, but there are so many moving parts that it's hard enough getting Unity and Flutter and Android Studio and XCode to work together without yet another moving part!

**Comment by juicycleff** [comment]
@jamesncl your actually right. I will revert it as soon as the doc website is ready. It will contain tested unity versions against the package versions

**Comment by jamesncl** [comment]
@juicycleff  I was thinking about this, I'm not sure if you can revert this? Now that a version with the major number 2022 has been published, can you actually go back to a smaller number? I don't know, I've never tried to do that myself, just wondered if it might actually cause more problems / might be best to just stick with it now?

PS I've been using Unity 2022.1.2f1 and everything seems to work okay. When Unity hits the next LTS release (probably 2022.3 or 2022.4) I suggest sticking with that - LTS releases are more stable and less likely to randomly break the package

**Comment by juicycleff** [comment]
@jamesncl yeah I thought of that too. Quite frankly I will break the versioning matching unity but keep the ordering from the last release. But thanks for the docs you wrote will add it to the docs. If you like I can give you admin rights in this repo too

**Comment by jamesncl** [comment]
@juicycleff thanks :) Okay if I go through the issues list and close some old ones?

**Comment by juicycleff** [comment]
> @juicycleff thanks :) Okay if I go through the issues list and close some old ones?

Yes please you can

**Comment by Ahmadre** [comment]
I added some docs today :3 please consider adding them to the docs page :3
